docs/jdbc.md

```diff
@@ -0,0 +1,447 @@
+```
+
+**Note:** The host portion of the JDBC URL is currently unused, and has no
+effect on the connection process. The SocketFactory will get your instances IP
+address based on the provided `cloudSqlInstance` arg.
+
+#### SQL Server
+
+Base JDBC URL: `jdbc:sqlserver://localhost;databaseName=<DATABASE_NAME>`
+
+SOCKET_FACTORY_CLASS: `com.google.cloud.sql.sqlserver.SocketFactory`
+
+The full JDBC URL should look like this:
+
+```
+jdbc:sqlserver://localhost;databaseName=<DATABASE_NAME>;socketFactoryClass=com.google.cloud.sql.sqlserver.SocketFactory;socketFactoryConstructorArg=<INSTANCE_CONNECTION_NAME>;user=<USER_NAME>;password=<PASSWORD>
+```
+
+**Note:** The host portion of the JDBC URL is currently unused, and has no
+effect on the connection process. The SocketFactory will get your instances IP
+address based on the provided `socketFactoryConstructorArg` arg.
+
+### Specifying IP Types
+
+The `ipTypes` argument is used to specify a preferred order of IP types used
+to connect via a comma delimited list. For example, `ipTypes=PUBLIC,PRIVATE`
+will use the instance's Public IP if it exists, otherwise private. The
+value `ipTypes=PRIVATE` will force the Cloud SQL instance to connect via
+it's private IP. The value `ipTypes=PSC` will force the Cloud SQL instance to
+connect to the database
+via [Private Service Connect](https://cloud.google.com/vpc/docs/private-service-connect).
+If not specified, the default used is `ipTypes=PUBLIC,PRIVATE`.
+
+For more info on connecting using a private IP address,
+see [Requirements for Private IP](https://cloud.google.com/sql/docs/mysql/private-ip#requirements_for_private_ip).
+
+#### Mysql
+
+```
+jdbc:mysql:///<DATABASE_NAME>?cloudSqlInstance=<INSTANCE_CONNECTION_NAME>&ipTypes=PRIVATE&socketFactory=com.google.cloud.sql.mysql.SocketFactory&user=<MYSQL_USER_NAME>&password=<MYSQL_USER_PASSWORD>
+```
+
+#### Maria DB
+
+```
+jdbc:mariadb://ignoreme:1234/<DATABASE_NAME>?cloudSqlInstance=<INSTANCE_CONNECTION_NAME>&ipTypes=PRIVATE&socketFactory=com.google.cloud.sql.mariadb.SocketFactory&user=<MYSQL_USER_NAME>&password=<MYSQL_USER_PASSWORD>
+```
+
+#### Postgres
+
+```
+jdbc:postgresql:///<DATABASE_NAME>?cloudSqlInstance=<INSTANCE_CONNECTION_NAME>&ipTypes=PRIVATE&socketFactory=com.google.cloud.sql.postgres.SocketFactory&user=<POSTGRESQL_USER_NAME>&password=<POSTGRESQL_USER_PASSWORD>
+```
+
+#### SQL Server
+
+IP types can be specified by appending the ipTypes argument
+to `socketFactoryConstructorArg` using query syntax, such as:
+
+```
+jdbc:sqlserver://localhost;databaseName=<DATABASE_NAME>;socketFactoryClass=com.google.cloud.sql.sqlserver.SocketFactory;socketFactoryConstructorArg=<INSTANCE_CONNECTION_NAME>?ipTypes=PRIVATE;user=<USER_NAME>;password=<PASSWORD>
+```
+
+Or in java code:
+
+```java
+String jdbcURL=String.format("jdbc:sqlserver://localhost;databaseName=%s","<DATABASE_NAME>");
```

Comment:
This block isn't correctly indented.

Since this is all in a code fence, we don't need to add four spaces in front of everything.